### PR TITLE
[Vue] migrate date-picker and other period-selector related components

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of https://github.com/matomo-org/matomo/pull/18193.**

Changes:

* Migrate piwik-date-picker component to Vue.
* Migrate piwik-date-range-picker component.
* Migrate piwik-period-date-picker component.
* Migrate piwik-expand-on-click directive and piwik-expand-on-hover directive.
* Add event mapping support to createAngularJsAdapter().

Notes:

* Using the composition API (doing a lot of work in `setup()`) vs. the options API (properties in `defineComponent(...)`) moreso here since it's easier when watching multiple properties at the same time (though it's still hard to do w/ Vue). Neither API is preferred over the other by the vue apparently.

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
@sgiehl looks like tests are passing, this one should be ready to review.

**Comment by diosmosis** [comment]
Actually, there's still one issue here, I'd wait a bit.

**Comment by diosmosis** [comment]
@sgiehl this is ready for review now.

**Comment by peterhashair** [comment]
that looks good to me.

File: plugins/CoreHome/vue/src/DatePicker/DatePicker.vue [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+    function viewDateChanged(): boolean {
+      let date = props.viewDate;
+      if (!date) {
+        return false;
+      }
+
+      if (!(date instanceof Date)) {
+        try {
+          date = $.datepicker.parseDate('yy-mm-dd', date);
```
Comment (sgiehl) [inline plugins/CoreHome/vue/src/DatePicker/DatePicker.vue]: any reason for not using `Periods/utilities/parseDate`? I know it's handling some stuff more than needed, but might be good to minimize the direct usage of the datepicker methods, so it's easier to switch it at some point maybe.

**Review by sgiehl — APPROVED** [review]
Left a minor comment. Besides that the code looks fine and the tests seems still to pass. Feel free to merge @diosmosis

File: plugins/CoreHome/vue/src/DatePicker/DatePicker.vue (hunk 1 above)
Comment (diosmosis) [reply plugins/CoreHome/vue/src/DatePicker/DatePicker.vue]: No reason, I missed this. You're right it should use the typescript method.

**Pull request merged**